### ISSUE TITLE
data.json just shows the first 100 values of attributes

### ISSUE THREAD

**Issue by AimbotNooby** [maptiler/tileserver-gl#1428]
Hey, when I retrieve the /data/mbtiles.json, /data.json or index.json, I only get the first 100 values of the attributes I use for my LineStrings, even if the count is over 100.
Is there a way to configure how much values are shown in the json? I couldn't find anything in the documentation.

> ...
    "tilestats": {
        "layerCount": 1,
        "layers": [
            {
                "layer": "main",
                "count": 297,
                "geometry": "LineString",
                "attributeCount": 12,
                "attributes": [                         
                    {
                        "attribute": "date",
                        "count": **297**,
                        "type": "string",
                        "values": [
                            "2023-12-01",
                            "2023-12-02",
                            "2023-12-03",
                            "2023-12-04",
                            "2023-12-05",
                            "2023-12-06",
                            "2023-12-07",
                            "2023-12-08",
                            "2023-12-09",
                            "2023-12-11",
                            "2023-12-12",
                            "2023-12-13",
                            "2023-12-14",
                            "2023-12-15",
                            "2023-12-16",
                            "2023-12-17",
                            "2023-12-18",
                            "2023-12-19",
                            "2023-12-20",
                            "2023-12-21",
                            "2023-12-22",
                            "2023-12-23",
                            "2023-12-24",
                            "2023-12-25",
                            "2023-12-26",
                            "2023-12-27",
                            "2023-12-28",
                            "2023-12-29",
                            "2023-12-30",
                            "2023-12-31",
                            "2024-01-03",
                            "2024-01-04",
                            "2024-01-05",
                            "2024-01-06",
                            "2024-01-07",
                            "2024-01-08",
                            "2024-01-10",
                            "2024-01-11",
                            "2024-01-12",
                            "2024-01-13",
                            "2024-01-14",
                            "2024-01-15",
                            "2024-01-16",
                            "2024-01-17",
                            "2024-01-18",
                            "2024-01-19",
                            "2024-01-20",
                            "2024-01-21",
                            "2024-01-22",
                            "2024-01-23",
                            "2024-01-24",
                            "2024-01-28",
                            "2024-01-29",
                            "2024-01-31",
                            "2024-02-01",
                            "2024-02-02",
                            "2024-02-03",
                            "2024-02-04",
                            "2024-02-05",
                            "2024-02-06",
                            "2024-02-07",
                            "2024-02-08",
                            "2024-02-09",
                            "2024-02-10",
                            "2024-02-11",
                            "2024-02-12",
                            "2024-02-13",
                            "2024-02-14",
                            "2024-02-17",
                            "2024-02-18",
                            "2024-02-19",
                            "2024-02-20",
                            "2024-02-21",
                            "2024-02-22",
                            "2024-02-23",
                            "2024-02-26",
                            "2024-02-27",
                            "2024-02-29",
                            "2024-03-01",
                            "2024-03-02",
                            "2024-03-03",
                            "2024-03-05",
                            "2024-03-06",
                            "2024-03-07",
                            "2024-03-08",
                            "2024-03-09",
                            "2024-03-10",
                            "2024-03-11",
                            "2024-03-15",
                            "2024-03-16",
                            "2024-03-17",
                            "2024-03-18",
                            "2024-03-19",
                            "2024-03-20",
                            "2024-03-21",
                            "2024-03-22",
                            "2024-03-23",
                            "2024-03-24",
                            "2024-03-25",
                            "2024-03-28"
                        ]
                    }
                ]
            }
        ]
    },
    "tilejson": "2.0.0"
}

**Comment by acalcutt** [comment]
Is it the same on the index page? are you able to access the files that arn't listed there at all?

I am not aware of a parameter or limit... we would have to look around the code and see where that limit may come from

**Comment by AimbotNooby** [comment]
Yes it's the same on the index.json and yes I can access the files that arn't listet. The parameters with the correct values are also available in the not listed files.

**Comment by okimiko** [comment]
Is it possible that a request timeout was hit? How long does the query take and is it always limited to exactly 100?  By `LineString` I assume a geojson source?

**Comment by AimbotNooby** [comment]
The query takes around 200ms. 
The limit is always exactly 100.
The Tileserver serves mbtiles, which are generated by tippecanoe from a geojson.

**Comment by okimiko** [comment]
@AimbotNooby I have been down in the rabbit hole and found some batch and chunk sizes on my way down, but everything was  fine. The "issue" is hard coded here:
https://github.com/felt/tippecanoe/blob/423fea15448386ccc832686f55f072b1f97f73ba/mbtiles.cpp#L27

I could verify it with my own geojson file /o\
